src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/SharedObjects.cs

```diff
@@ -0,0 +1,627 @@
+                    AzureDevOpsBuildDefinitionId = AzureDevOpsBuildDefinitionId1
+                }
+            };
+
+        public static readonly List<BuildData> ExpectedDuplicatedAssetsBuildData = new List<BuildData>()
+        {
+            new BuildData(Commit, AzureDevOpsAccount1, AzureDevOpsProject1, AzureDevOpsBuildNumber1, AzureDevOpsRepository1, AzureDevOpsBranch1, false, false)
+            {
+                GitHubBranch = AzureDevOpsBranch1,
+                GitHubRepository = "https://github.com/dotnet/arcade",
+                Assets = ExpectedAssets1,
+                AzureDevOpsBuildId = AzureDevOpsBuildId1,
+                AzureDevOpsBuildDefinitionId = AzureDevOpsBuildDefinitionId1
+            },
+            new BuildData(Commit, AzureDevOpsAccount1, AzureDevOpsProject1, AzureDevOpsBuildNumber1, AzureDevOpsRepository1, AzureDevOpsBranch1, false, false)
+            {
+                GitHubBranch = AzureDevOpsBranch1,
+                GitHubRepository = "https://github.com/dotnet/arcade",
+                Assets = ExpectedAssets1,
+                AzureDevOpsBuildId = AzureDevOpsBuildId1,
+                AzureDevOpsBuildDefinitionId = AzureDevOpsBuildDefinitionId1
+            }
+        };
+        #endregion
+
+        public static void CompareManifestBuildData(ManifestBuildData actual, ManifestBuildData expected)
+        {
+            actual.AzureDevOpsAccount.Should().Be(expected.AzureDevOpsAccount);
+            actual.AzureDevOpsBranch.Should().Be(expected.AzureDevOpsBranch);
+            actual.AzureDevOpsBuildDefinitionId.Should().Be(expected.AzureDevOpsBuildDefinitionId);
+            actual.AzureDevOpsBuildId.Should().Be(expected.AzureDevOpsBuildId);
+            actual.AzureDevOpsBuildNumber.Should().Be(expected.AzureDevOpsBuildNumber);
+            actual.AzureDevOpsProject.Should().Be(expected.AzureDevOpsProject);
+            actual.AzureDevOpsRepository.Should().Be(expected.AzureDevOpsRepository);
+            actual.InitialAssetsLocation.Should().Be(expected.InitialAssetsLocation);
+            actual.PublishingVersion.Should().Be(expected.PublishingVersion);
+        }
+
+        public static void CompareSigningInformation(SigningInformation actualSigningInfo, SigningInformation expectedSigningInfo)
+        {
+            actualSigningInfo.CertificatesSignInfo.Should().BeEquivalentTo(expectedSigningInfo.CertificatesSignInfo);
+            actualSigningInfo.FileExtensionSignInfos.Should().BeEquivalentTo(expectedSigningInfo.FileExtensionSignInfos);
+            actualSigningInfo.FileSignInfos.Should().BeEquivalentTo(expectedSigningInfo.FileSignInfos);
+            actualSigningInfo.ItemsToSign.Should().BeEquivalentTo(expectedSigningInfo.ItemsToSign);
+        }
+
+        public static void CompareBuildDataInformation(BuildData actualBuildData, BuildData expectedBuildData)
+        {
+            actualBuildData.Assets.Should().BeEquivalentTo(expectedBuildData.Assets);
```

Comment:
Is there some reason this isn't just actualBuildData.Should().BeEquivalentTo(expectedBuildData)? (Which also means we don't need the method anymore)

Reply:
Because the comparison always returns false & the comparer used in product code doesn't check everything, just the surface metadata, so it's not helpful here.

Reply:
BeEquivalentTo should do deep tree comparison https://fluentassertions.com/objectgraphs/.  Is it not working?

Reply:
It didn't seem to be working, no. I'll give it another shot to make sure that there wasn't something else causing them to fail (and check the options while I'm there).

Reply:
I must have been using the wrong assert, it's all good now with no tweaks needed. :)